core/products/future.go

```diff
@@ -209,24 +203,45 @@ func (f *Future) updateSettlementData(ctx context.Context, data oracles.OracleDa
 		f.log.Debug("new oracle data received", data.Debug()...)
 	}
 
-	settlementData, err := data.GetUint(f.oracle.binding.settlementDataProperty)
-	if err != nil {
-		f.log.Error(
-			"could not parse the property acting as settlement data",
-			logging.Error(err),
-		)
-		return err
+	odata := &oracleData{
+		settlData: &num.Numeric{},
+	}
+	switch f.oracle.binding.settlementDataPropertyType {
+	case datapb.PropertyKey_TYPE_DECIMAL:
+		settlDataAsDecimal, err := data.GetDecimal(f.oracle.binding.settlementDataProperty)
+		if err != nil {
+			f.log.Error(
+				"could not parse decimal type property acting as settlement data",
+				logging.Error(err),
+			)
+			return err
+		}
+
+		odata.settlData.SetDecimal(&settlDataAsDecimal)
+		f.oracle.binding.settlementDataDecimals = uint64(0)
```

Comment:
?? why are we setting this to 0 here? If we're dealing with a decimal, at this point we should've already made sure this field is valid (ie set to zer)

Reply:
Yes, this is a mistake, sorry.
Corrected.